apps/worker/src/monitor/tx.status.monitor.service.ts

```diff
@@ -0,0 +1,130 @@
+import { InjectRedis } from '@liaoliaots/nestjs-redis';
+import { Processor, WorkerHost, OnWorkerEvent, InjectQueue } from '@nestjs/bullmq';
+import { Injectable, Logger, OnApplicationBootstrap, OnModuleDestroy } from '@nestjs/common';
+import { Job, Queue } from 'bullmq';
+import Redis from 'ioredis';
+import { EventEmitter2 } from '@nestjs/event-emitter';
+import { MILLISECONDS_PER_SECOND } from 'time-constants';
+import { BlockHash, Hash } from '@polkadot/types/interfaces';
+import { map, tap, timeout } from 'rxjs';
+import { BlockchainService } from '../../../../libs/common/src/blockchain/blockchain.service';
+import { ConfigService } from '../../../../libs/common/src/config/config.service';
+import { ITxMonitorJob } from '../interfaces/status-monitor.interface';
+import { QueueConstants } from '../../../../libs/common/src';
+import { SECONDS_PER_BLOCK } from '../../../../libs/common/src/constants';
+
+@Injectable()
+@Processor(QueueConstants.TRANSACTION_RECEIPT_QUEUE_NAME, {
+  concurrency: 2,
+})
+export class TxStatusMonitoringService extends WorkerHost implements OnApplicationBootstrap, OnModuleDestroy {
+  private logger: Logger;
+
+  constructor(
+    @InjectRedis() private cacheManager: Redis,
+    @InjectQueue(QueueConstants.TRANSACTION_RECEIPT_QUEUE_NAME) private txReceiptQueue,
+    @InjectQueue(QueueConstants.PUBLISH_QUEUE_NAME) private publishQueue: Queue,
+    private blockchainService: BlockchainService,
+    private configService: ConfigService,
+    private eventEmitter: EventEmitter2,
+  ) {
+    super();
+    this.logger = new Logger(this.constructor.name);
+  }
+
+  public async onApplicationBootstrap() {
+    this.logger.debug('Starting publishing service');
+  }
+
+  public onModuleDestroy() {
+    try {
+      this.logger.debug('Shutting down publishing service');
+    } catch (e) {
+      // 🐂 //
+    }
+  }
+
+  async process(job: Job<ITxMonitorJob, any, string>): Promise<any> {
+    this.logger.log(`Monitoring job ${job.id} of type ${job.name}`);
+    try {
+      const numberBlocksToParse = 100n;
```

Comment:
If not a configurable param I think it would be better if we put this constrant into `BlockchainConstants` which is in common module

Reply:
Moved it constants for this PR, I would take the mortality from tx but havent found how, so keeping it const for now